x-pack/plugins/security/server/authentication/providers/basic.ts

```diff
@@ -0,0 +1,213 @@
+  }
+}
+
+/**
+ * Describes the parameters that are required by the provider to process the initial login request.
+ */
+interface ProviderLoginAttempt {
+  username: string;
+  password: string;
+}
+
+/**
+ * The state supported by the provider.
+ */
+interface ProviderState {
+  /**
+   * Content of the HTTP authorization header (`Basic base-64-of-username:password`) that is based
+   * on user credentials used at login time and that should be provided with every request to the
+   * Elasticsearch on behalf of the authenticated user.
+   */
+  authorization?: string;
+}
+
+/**
+ * Provider that supports request authentication via Basic HTTP Authentication.
+ */
+export class BasicAuthenticationProvider extends BaseAuthenticationProvider {
+  /**
+   * Performs initial login request using username and password.
+   * @param request Request instance.
+   * @param attempt User credentials.
+   * @param [state] Optional state object associated with the provider.
+   */
+  public async login(
+    request: KibanaRequest,
+    attempt: ProviderLoginAttempt,
+    state?: ProviderState | null
+  ) {
+    this.logger.debug('Trying to perform a login.');
+
+    if (!attempt || !attempt.username || !attempt.password) {
+      this.logger.debug('Username and/or password not provided.');
+      return AuthenticationResult.notHandled();
+    }
+
+    try {
+      const authHeaders = {
+        authorization: `Basic ${Buffer.from(`${attempt.username}:${attempt.password}`).toString(
+          'base64'
+        )}`,
+      };
+      const user = await this.getUser(request, authHeaders);
+
+      this.logger.debug('Login has been successfully performed.');
+      return AuthenticationResult.succeeded(user, { authHeaders, state: authHeaders });
+    } catch (err) {
+      this.logger.debug(`Failed to perform a login: ${err.message}`);
+      return AuthenticationResult.failed(err);
+    }
+  }
+
+  /**
+   * Performs request authentication using Basic HTTP Authentication.
+   * @param request Request instance.
+   * @param [state] Optional state object associated with the provider.
+   */
+  public async authenticate(request: KibanaRequest, state?: ProviderState | null) {
+    this.logger.debug(`Trying to authenticate user request to ${request.url.path}.`);
+
+    // try header-based auth
+    const {
+      authenticationResult: headerAuthResult,
+      headerNotRecognized,
+    } = await this.authenticateViaHeader(request);
+    if (headerNotRecognized) {
+      return headerAuthResult;
+    }
+
+    let authenticationResult = headerAuthResult;
+    if (authenticationResult.notHandled() && state) {
+      authenticationResult = await this.authenticateViaState(request, state);
+    } else if (authenticationResult.notHandled() && canRedirectRequest(request)) {
+      // If we couldn't handle authentication let's redirect user to the login page.
+      const nextURL = encodeURIComponent(
+        `${this.options.basePath.get(request)}${request.url.path}`
```

Comment:
Hmm, don't have strong opinion, the call like `.basePath.prepend(request, request.url.path)` may look a bit weird though...

Reply:
with [Handler interface](https://github.com/elastic/kibana/pull/36509) it will look like
```js
basePath.prepend(url: string) => string
```
I'm not insist. we can keep it as is

Reply:
Ah, right, once we have this shiny Handler infterface implemented, that would look neat, yeah.

Reply:
can wait until it's implemented